analyzer/validatorstakinghistory/validatorstakinghistory.go

```diff
@@ -171,6 +182,14 @@ func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch,
 			acct.Escrow.Debonding.TotalShares,
 			len(delegations),
 		)
+		// Update staking rewards for this validator in the previous epoch.
+		if ev, exists := stakingRewards[addr]; exists {
+			batch.Queue(queries.ValidatorStakingRewardUpdate,
+				vID.String(),
+				epoch.epoch-1, // previous epoch
```

Comment:
can you affirm that this is all right

1. staking rewards appear in the first block of the next epoch

Reply:
Hm I asked a follow up in core-platform but as of now, the e2e integration tests show that the staking rewards are indeed populated in the first block of the next epoch. However, I'm not sure if there are any failure edge cases where the rewards might be populated in a subsequent block; will ask around.